l3packages/xparse/xparse.dtx

```diff
@@ -1442,7 +1442,22 @@
   {
     \__kernel_cmd_if_xparse:NTF #1
       {
-        \tl_set:Ne \ArgumentSpecification { \tl_item:Nn #1 { 2 } }
+        \tl_set:Ne \ArgumentSpecification
+          {
+            \exp_args:No \tl_if_head_eq_meaning:nNTF {#1} \@@_start_optimized:
+              {
+                \prg_replicate:nn
+                  {
+                    \str_count:e
+                      {
+                        \exp_args:Nc \cs_parameter_spec:N
+                          { \cs_to_str:N #1 \c_space_tl code }
```

Comment:
Not sure which way is better, making use of the implementation detail of `\<cmdname>~code` made from `\<cmdname>`, or the content of optimized command, in which the second (and last) token is exactly `\<cmdname>~code`.

Using the second way, we can get `\cs_parameter_spec:N \<cmdname>~code` from
```tex
\exp_last_unbraced:NNo \use_i:nn
  \cs_parameter_spec:N \mycmda
```

Reply:
We can rely on implementation here - this is code that is clearly part of  `ltcmd`/`xparse`.